wavefront_pyformance/wavefront_reporter.py

```diff
@@ -0,0 +1,143 @@
+                                                reporting_interval=reporting_interval,
+                                                clock=clock)
+        self.source = source
+        self.prefix = prefix
+        tags = tags or {}
+        self.tagStr = ' '.join(['\"%s\"=\"%s\"' % (k,v) for (k,v) in tags.items()])
+
+    def report_now(self, registry=None, timestamp=None):
+        raise NotImplementedError('use WavefrontProxyReporter or WavefrontDirectReporter')
+
+    def _collect_metrics(self, registry, timestamp=None):
+        metrics = registry.dump_metrics()
+        metrics_data = []
+        for key in metrics.keys():
+            is_delta = delta.is_delta_counter(key, registry)
+            for value_key in metrics[key].keys():
+                if is_delta:
+                    registry.counter(key).dec(metrics[key][value_key]) # decrement delta counter
+                metric_name = self._get_metric_name(self.prefix, key, value_key, is_delta=is_delta)
+                metric_line = self._get_metric_line(metric_name, metrics[key][value_key], timestamp)
+                metrics_data.append(metric_line)
+        return metrics_data
+
+    def _get_metric_line(self, name, value, timestamp):
+        if timestamp:
+            return "%s %s %s source=\"%s\" %s" % (name, value, timestamp, self.source, self.tagStr)
+        else:
+            return "%s %s source=\"%s\" %s" % (name, value, self.source, self.tagStr)
+
+    def _get_metric_name(self, prefix, name, value_key, is_delta=False):
+        return delta.get_delta_name(prefix, name, value_key) if is_delta else "%s%s.%s" % (prefix, name, value_key)
+
+
+class WavefrontProxyReporter(WavefrontReporter):
+    """
+    This reporter requires a host and port to report data to a Wavefront proxy.
+    """
+
+    def __init__(self, host, port=2878, source="wavefront-pyformance", registry=None, reporting_interval=10, clock=None,
+                 prefix="proxy.", tags={}):
+        super(WavefrontProxyReporter, self).__init__(source=source,
+                                                     registry=registry,
+                                                     reporting_interval=reporting_interval,
+                                                     clock=clock,
+                                                     prefix=prefix,
+                                                     tags=tags)
+        self.host = host
+        self.port = port
+        self.socket_factory = socket.socket
+        self.proxy_socket = None
+
+    def report_now(self, registry=None, timestamp=None):
+        timestamp = timestamp or int(round(self.clock.time()))
+        metrics = self._collect_metrics(registry or self.registry, timestamp)
+        if metrics:
+            try:
+                if not self.proxy_socket:
+                    self._connect()
+                for line in metrics:
+                    self.proxy_socket.send(line.encode('utf-8') + "\n")
```

Comment:
We might need to use a ReconnectingSocket (https://github.com/wavefrontHQ/java/blob/c6f8552e78f9a04c873fc993880c74477c26037b/java-lib/src/main/java/com/wavefront/metrics/ReconnectingSocket.java) equivalent in Python